modules/apps/locked-items/locked-items-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -0,0 +1,43 @@
+<%--
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<%
+LockedItemsDisplayContext lockedItemsDisplayContext = (LockedItemsDisplayContext)request.getAttribute(LockedItemsDisplayContext.class.getName());
+
+LockedItemsScreen lockedItemsScreen = lockedItemsDisplayContext.getLockedItemsScreen();
+%>
+
+<clay:container-fluid
+	cssClass="container-view"
+>
+	<clay:row>
+		<clay:col
+			lg="3"
+		>
+			<clay:vertical-nav
+				verticalNavItems="<%= lockedItemsDisplayContext.getVerticalNavItemList() %>"
+			/>
+		</clay:col>
+
+		<clay:col
+			lg="9"
+		>
+			<clay:sheet
+				cssClass="full"
+			>
+				<clay:sheet-header>
+					<p class="sheet-title"><%= lockedItemsScreen.getName(locale) %></p>
+					<p class="sheet-text"><%= lockedItemsScreen.getDescription(locale) %></p>
+				</clay:sheet-header>
+
+				<liferay-util:include page="<%= lockedItemsScreen.getJspPath() %>" servletContext="<%= lockedItemsScreen.getServletContext() %>" />
```

Comment:
The usage of the `include` tag forces the `LockedItemScreen` interface to expose too much stuff (e.g. servlet context), and forces implementations to use a JSP. Can you try to use the same approach as the `Renderer` (asset framework) or `DLPreviewRenderer`? Instead of exposing all the information required by the include tag, simply provide a `render` method that does whatever it has to do (including setting any request attributes).

With that, you should be able to remove at least the `getJsp`, `getServletContext` and `setAttribute` methods from the interface.